dotnetv3/RDS/Scenarios/RDSInstanceScenario/RDSInstanceScenario.cs

```diff
@@ -0,0 +1,485 @@
+    {
+        Console.WriteLine(sepBar);
+        // Get a list of allowed engines.
+        var allowedEngines =
+            await rdsWrapper.DescribeDBEngineVersions(engine, dbParameterGroupFamily);
+
+        Console.WriteLine($"Available DB engine versions for parameter group family {dbParameterGroupFamily}:");
+        int i = 1;
+        foreach (var version in allowedEngines)
+        {
+            Console.WriteLine(
+                $"\t{i}. Engine: {version.Engine} Version {version.EngineVersion}.");
+            i++;
+        }
+
+        var choiceNumber = 0;
+        while (choiceNumber < 1 || choiceNumber > allowedEngines.Count)
+        {
+            Console.WriteLine("8. Select an available DB engine version by entering a number from the list above:");
+            var choice = Console.ReadLine();
+            Int32.TryParse(choice, out choiceNumber);
+        }
+
+        var engineChoice = allowedEngines[choiceNumber - 1];
+        Console.WriteLine(sepBar);
+        return engineChoice;
+    }
+
+    /// <summary>
+    /// Choose a DB instance class for a particular engine and engine version.
+    /// </summary>
+    /// <param name="engine">DB engine for instance choice.</param>
+    /// <param name="engineVersion">DB engine version for instance choice.</param>
+    /// <returns>The selected orderable DB instance option.</returns>
+    public static async Task<OrderableDBInstanceOption> ChooseDbInstanceClass(string engine, string engineVersion)
+    {
+        Console.WriteLine(sepBar);
+        // Get a list of allowed instance classes.
+        var allowedInstances =
+            await rdsWrapper.DescribeOrderableDBInstanceOptions(engine, engineVersion);
+
+        Console.WriteLine($"8. Available micro DB instance classes for engine {engine} and version {engineVersion}:");
+        int i = 1;
+
+        // Filter to micro instances for this example.
+        allowedInstances = allowedInstances
+            .Where(i => i.DBInstanceClass.Contains("micro")).ToList();
+
+        foreach (var instance in allowedInstances)
+        {
+            Console.WriteLine(
+                $"\t{i}. Instance class: {instance.DBInstanceClass} (storage type {instance.StorageType})");
+            i++;
+        }
+
+        var choiceNumber = 0;
+        while (choiceNumber < 1 || choiceNumber > allowedInstances.Count)
+        {
+            Console.WriteLine("9. Select an available instance class by entering a number from the list above:");
+            var choice = Console.ReadLine();
+            Int32.TryParse(choice, out choiceNumber);
+        }
+
+        var instanceChoice = allowedInstances[choiceNumber - 1];
+        Console.WriteLine(sepBar);
+        return instanceChoice;
+    }
+
+    /// <summary>
+    /// Create a new RDS DB instance.
+    /// </summary>
+    /// <param name="parameterGroup">Parameter group to use for the instance.</param>
```

Comment:
I think we have to always qualify "instance" as "DB instance". You can wait for editorial comment to be sure if you want.

Reply:
You're correct, it even says so in the service names guide. I fixed it throughout.